### ISSUE TITLE
Removing a TeamUser does not remove SharePoint Online permissions.

### ISSUE THREAD

**Issue by evandimsby** [MicrosoftDocs/office-docs-powershell#4519]
Whether using PS or the Teams client, removing a member from a Team will only remove the team from the previous member's list of Teams. 

However, the user is still able to access the SPO site and all contents even though they are not actually a member of the team anymore.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b8d16f25-be6f-06eb-2cd7-9e64eab124e0
* Version Independent ID: a8181f1a-e575-410a-d5a4-0214c427b92d
* Content: [Remove-TeamUser (teams)](https://docs.microsoft.com/en-us/powershell/module/teams/remove-teamuser?view=teams-ps)
* Content Source: [teams/teams-ps/teams/Remove-TeamUser.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/teams/teams-ps/teams/Remove-TeamUser.md)
* Service: **teams-powershell**
* GitHub Login: @kenwith
* Microsoft Alias: **kenwith**

**Comment by kenwith** [comment]
@evandimsby - thank you for the feedback on this cmdlet. I know it takes time to submit an issue or create a pull request as you are gaining the knowledge learning and using the products. And I appreciate it.

This issue list is for the docs and we don't have a way to provide any type of support. We read every issue and always want to do anything we can to help. It does seem like a bug or something weird going on because it doesn't make sense that if a user was removed from the team that they should still be able to access the SPO of the team. Or if it is something by design it is definitely confusing and we need to update the docs around this area.

You can open a support ticket in the admin center of your tenant and they can determine right away what is going on and escalate to the appropriate product team. I know the product team listens to all feedback and I wish I had a better way to open these bugs but I don't have the experience or resources of the support engineers so I think that would be the best route.

Some things I have noticed with Teams is that a change can take a long time, up to 24 hours, to propagate on things. I don't know if that could be the case but support engineering would know.

I think we will need to update this reference article at any rate and add better information once we get this figured out. I will also ping a contact I have on the Teams product group in parallel (and support can open an official channel and escalate for you immediately so that is always the better route). 

If you wouldn't mind keeping me in the loop I would really appreciate it. Also, if you are interested in contributing to the docs and sharing your knowledge I can guide you on submitting a Pull Request and will get it validated and merged.

**Comment by kenwith** [comment]
I also found this one in User Voice and up voted it.. https://microsoftteams.uservoice.com/forums/908686-bug-reports/suggestions/37649128-deleted-member-still-on-my-sharepoint

**Comment by evandimsby** [comment]
> 
> 
> I also found this one in User Voice and up voted it.. https://microsoftteams.uservoice.com/forums/908686-bug-reports/suggestions/37649128-deleted-member-still-on-my-sharepoint

Good find! I voted on it, too. I would imagine once more people start noticing this loophole that it should get attention fairly quickly. I did open a support request and the techs were rather puzzled. Even as Global SP Admin, i cannot actually view the memberships of SPO groups that are created via Teams, and therefore, not able to verify/revoke the access.

**Comment by yogkumgit** [comment]
@officedocsbot assign @yogkumgit

**Comment by tseward** [comment]
Hi folks,

When a user is removed from a Microsoft Team, they are removed from the Office 365 Group which also governs access to the SharePoint Online site.

However, SharePoint Online also has it's own ACL controls via the way of direct assignment, sharing links, and SharePoint Groups (which may have the user directly added or nested via an Azure AD security group, mail-enabled security group, or Active Directory synchronized security group). Removing the user from the Office 365 Group does not impact these forms of access control, thus the user may continue having access to the SharePoint Online site.

If the Office 365 Group (Team) is Public, the user may also re-add themselves at any time which would in turn give them access to the SharePoint Online site with Edit rights.

**Comment by kenwith** [comment]
I also found out there is a documentation effort underway that @MikePlumleyMSFT and @BrendaCarter are leading to discuss decommissioning teams and I pointed this info to them as a key area to note. Thanks all for this feedback.

**Comment by LinkMJB** [comment]
This should absolutely be addressed as its own issue, and goes much further than being a simple documentation issue. This is a huge problem. This creates a false sense of security, that a user was properly removed from accessing a Sharepoint site. @kenwith can this be escalated to appropriate teams?

**Comment by MikePlumleyMSFT** [comment]
@LinkMJB, @tseward's comment is correct. Because SharePoint has it's own permission model outside the scope of Teams, it's possible for users to have SharePoint site or direct file permissions that aren't related to the site's relationship with a team. You can limit this by [changing who can share files, folders, and the site[(https://support.microsoft.com/office/94b26e0b-2822-49d4-929a-8455698654b3).
